talks/storing_data/index.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Storing data"
+author: "Florian Ziemen, Tobi Kölling, Lukas Kluft"
+subtitle: ""
+format:
+  revealjs:
+    transition: slide
+    slide-number: true
+    code-background: true
+    logo: /static/logos.png
+    footer: "[Talks page](/talks)"
+    theme: [default, ../custom.scss]
+    chalkboard:
+      buttons: true
+    navigation-mode: linear
+    auto-stretch: true
+    center: true
+    mermaid:
+      theme: default
+---
+
+# Datasets
+
+## Datasets are
+
+![(from [xarray docs](https://docs.xarray.dev/en/stable/user-guide/data-structures.html))](dataset.png)
+
+
+:::: {.columns}
+
+::: {.column width="50%"}
+* n-dimensional variables
+* shared dimensions
+:::
+
+::: {.column width="50%"}
+* coordinates
+* attributes for metadata
+:::
+
+::::
+
+## Datasets are not
+
+* a single file
+* a storage format
+* shaped by storage & handling
+
+::: {.notes}
+In the past, people often designed datasets to match their file handling requirements.
+The resulting view on a dataset seems to be overly restrictive and leads to unfortunate usage patterns.
+:::
+
+# Storage and access
+
+## Access to subsets
+
+:::: {.columns}
+
+::: {.column width="50%"}
+![](data_usage_1.png)
+:::
+
+::: {.column width="50%"}
+![](data_usage_2.png)
+:::
+
+::::
+
+Without subset access and hierarchies, analysis scripts are **forced** to load way too much data.
+
+
+## Align the ordering of data with read patterns
+
+![[https://www.unidata.ucar.edu/blogs/developer/entry/chunking_data_why_it_matters](https://www.unidata.ucar.edu/blogs/developer/entry/chunking_data_why_it_matters)](https://www.unidata.ucar.edu/blog_content/images/2013/blog_rew_chunking.png)
+
+Make a compromise that's okay-ish for everybody by chunking along all dimensions.
+
+## Compress small chunks
+
+* Any access will require uncompressing entire chunks. 
+* By keeping them small to reduce the amount of data that will be uncompressed but not used.
+* Keep them big enough for the compressor to do its job.
+* Usually MB-ish blocks are a good compromise.
+
+## Use fast compressors
+
+* [deflate](https://en.wikipedia.org/wiki/Deflate) (gzip/...) is widely used but slow.
+* [lz4](https://en.wikipedia.org/wiki/LZ4_(compression_algorithm)) and [zstd](https://en.wikipedia.org/wiki/Zstd) are much faster.
+* Use lz4 or zstd from [blosc](http://blosc.org/) as a good standard.
+
+
+# Mapping a dataset to files
+
+## Contents of a dataset
+
+* Metadata
+* n-d variables consisting of many numbers (floats)
+
+## Zarr is as simple as it gets.
+
+* .json files for the metadata
+* binary files for compressed chunks
+
+As chunks are stored separately, this scales for any size of dataset. We are working with a 500 TB dataset in nextGEMS.
```

Comment:
For the storyline it might be better to end on Zarr and start with NetCDF4.

1. NetCDF: the good old faithful, but file-centric approach leads to bad patterns
2. Kerchunk: helps to construct datasets from various sources
3. Zarr: building single large datasets in one go

Reply:
I decided to move the zarr behind the netCDF, but keep kerchunk in place, as it kind of unites both (and thus is probably easier to introduce after the two).